packages/clerk-js/src/core/resources/SignUp.ts

```diff
@@ -138,26 +138,40 @@ export class SignUp extends BaseResource implements SignUpResource {
   };
 
   attemptWeb3WalletVerification = async (params: AttemptWeb3WalletVerificationParams): Promise<SignUpResource> => {
-    const { generateSignature } = params || {};
-    if (!(typeof generateSignature === 'function')) {
-      clerkMissingOptionError('generateSignature');
-    }
+    const { signature, generateSignature } = params || {};
```

Comment:
🙃  To avoid the need for the `if/else` clause and the explicit `:string` type definition on `let verificationSignature: string;`

I would probably write this as:

```ts
  attemptWeb3WalletVerification = async (params: AttemptWeb3WalletVerificationParams): Promise<SignUpResource> => {
    const { signature, generateSignature } = params || {};
    if (signature) {
      return this.attemptVerification({ signature, strategy: 'web3_metamask_signature' });
    }

    if (typeof generateSignature !== 'function') {
      clerkMissingOptionError('generateSignature');
    }

    const { nonce } = this.verifications.web3Wallet;
    if (!nonce) {
      clerkVerifyWeb3WalletCalledBeforeCreate('SignUp');
    }

    const generatedSignature = await generateSignature({ identifier: this.web3wallet!, nonce });
    return this.attemptVerification({ signature: generatedSignature, strategy: 'web3_metamask_signature' });
  };
```

This is super minor so feel free to ignore!